### PR TITLE
Fix task filtering.

### PR DESCRIPTION
**Pull request by hjoliver**
@oliver-sanders - I think this might be a bug in your task filtering function, found too late in the day to fully investigate - can you take a quick look, and if you agree it's a bug I'll finish off the PR?

If both the hidden pool (partially satisfied prereqs) and main pool are passed to the function - which they are, from `TaskPool.filter_task_proxies()` - it seems to return two copies of matched tasks in cycle points that appear in both pools.

These changes partially address #xxxx
These changes close #xxxx
This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.

- [x] Appropriate tests are included (unit and/or functional).

- [x] No change log entry required (why? e.g. invisible to users).

- [x] No documentation update required.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Looks sensible to me, thanks.

**Comment by hjoliver** [comment]
Added a test that fails on master.

File: cylc/flow/id_match.py [hunk 1]
```diff
@@ -213,7 +215,7 @@ def filter_ids(
             if warn:
                 LOG.warning(f"No active tasks matching: {id_}")
         else:
-            _cycles.extend(cycles)
+            _cycles.extend(list(cycles))
```
Comment (oliver-sanders) [inline cylc/flow/id_match.py:218]: The intermediate list is not necessary:

```suggestion
            _cycles.extend(cycles)
```

**Review by oliver-sanders — APPROVED** [review]
LGTM, cheers.

**Pull request merged**